gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java

```diff
@@ -84,6 +83,33 @@ public boolean apply(@Nonnull KafkaTopic kafkaTopic) {
     }));
   }
 
+  /**
+   * A helper method that returns the canonical metric name for a kafka metric. A typical canonicalized metric name would
+   * be of the following format: "{metric-group}_{client-id}_{metric-name}".
+   * @param metricGroup the type of the Kafka metric e.g."consumer-fetch-manager-metrics", "consumer-coordinator-metrics" etc.
+   * @param metricTags any tags associated with the Kafka metric, typically include the kafka client id, topic name, partition number etc.
+   * @param metricName the name of the Kafka metric e.g. "records-lag-max", "fetch-throttle-time-max" etc.
+   * @return the canonicalized metric name.
+   */
+  public String canonicalMetricName(String metricGroup, Collection<String> metricTags, String metricName) {
```

Comment:
Shall we make it `protected` or `package private`?

Reply:
Will make the change.